server/src/utils/media.ts

```diff
@@ -781,7 +781,12 @@ export class VaapiSwDecodeConfig extends BaseHWConfig {
 
   getFilterOptions(videoStream: VideoStreamInfo) {
     const options = this.getToneMapping(videoStream);
-    options.push('hwupload=extra_hw_frames=64');
+    if (videoStream.codecName === VideoCodec.HEVC) {
+      options.push('format=nv12');
+      options.push('hwupload=extra_hw_frames=64');
+    } else {
+      options.push('hwupload=extra_hw_frames=64');
+    }
     if (this.shouldScale(videoStream)) {
       options.push(`scale_vaapi=${this.getScaling(videoStream)}:mode=hq:out_range=pc:format=nv12`);
     }
```

Comment:
Whether to set the pixel format isn't really related to whether the input is HEVC. Also,`scale_vaapi` actually takes care of the format too, so I think it's better to do something like this (not tested):
```suggestion
    if (this.shouldScale(videoStream)) {
      options.push('hwupload=extra_hw_frames=64', `scale_vaapi=${this.getScaling(videoStream)}:mode=hq:out_range=pc:format=nv12`);
    } else {
      options.push('format=nv12', 'hwupload=extra_hw_frames=64');
    }
```
The same is true of QSV, so it should receive the same change.